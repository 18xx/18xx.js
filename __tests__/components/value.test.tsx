import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Hexagon from '../../src/hexagon';
import Point from '../../src/point';

import Value, { ValueProps } from '../../src/components/value';

describe('Value', () => {
  describe('render', () => {
    const props: ValueProps = {
      amount: 10,
      hexagon: new Hexagon('east-west'),
    };

    describe('when a position is specified', () => {
      const subject: React.ReactElement<Value> = (
        <Value { ...props} position={new Point(1, 2)} />
      );

      it('renders something matching the snapshot', () => {
        expect(renderer.create(subject)).toMatchSnapshot();
      });
    });

    describe('when a position is not specified', () => {
      describe('when the hex uses an east-west orientation', () => {
        const subject: React.ReactElement<Value> = (
          <Value { ...props} />
        );

        it('renders something matching the snapshot', () => {
          expect(renderer.create(subject)).toMatchSnapshot();
        });
      });

      describe('when the hex uses a north-south orientation', () => {
        const subject: React.ReactElement<Value> = (
          <Value { ...props} hexagon={new Hexagon('north-south')} />
        );

        it('renders something matching the snapshot', () => {
          expect(renderer.create(subject)).toMatchSnapshot();
        });
      });
    });
  });
});
