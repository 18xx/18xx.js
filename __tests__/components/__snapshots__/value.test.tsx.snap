// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Value render when a position is not specified when the hex uses a north-south orientation renders something matching the snapshot 1`] = `
<g>
  <circle
    cx={113.6}
    cy={55.42562584220407}
    fill="white"
    r={9}
    stroke="#000"
    strokeWidth="2"
  />
  <text
    fill="black"
    fontSize="11"
    textAnchor="middle"
    x={113.6}
    y={59.42562584220407}
  >
    10
  </text>
</g>
`;

exports[`Value render when a position is not specified when the hex uses an east-west orientation renders something matching the snapshot 1`] = `
<g>
  <circle
    cx={55.42562584220407}
    cy={112}
    fill="white"
    r={9}
    stroke="#000"
    strokeWidth="2"
  />
  <text
    fill="black"
    fontSize="11"
    textAnchor="middle"
    x={55.42562584220407}
    y={116}
  >
    10
  </text>
</g>
`;

exports[`Value render when a position is specified renders something matching the snapshot 1`] = `
<g>
  <circle
    cx={1}
    cy={2}
    fill="white"
    r={9}
    stroke="#000"
    strokeWidth="2"
  />
  <text
    fill="black"
    fontSize="11"
    textAnchor="middle"
    x={1}
    y={6}
  >
    10
  </text>
</g>
`;
